lib/action_account.rb

```diff
@@ -63,11 +69,67 @@ def log_text
         error_activating: "There was an error activating the user's profile. Please try again.",
         past_eligibility: 'User is past the 30 day review eligibility.',
         missing_uuid: 'Error: Could not find user with that UUID',
+        user_suspended: 'User has been suspended',
+        user_reinstated: 'User has been reinstated',
+        user_already_suspended: 'User has already been suspended',
+        user_is_not_suspended: 'User is not suspended',
       }
     end
   end
 
-  class ReviewReject < ReviewAction
+  module UserActions
+    include LogBase
+
+    def perform_user_action(args:, config:, action:)
+      table = []
+      messages = []
+      uuids = args
+      table << %w[uuid status]
+
+      users = User.where(uuid: uuids).order(:uuid)
+      users.each do |user|
+        log_texts = []
+        if action == :suspend && user.suspended?
+          log_texts << log_text[:user_already_suspended]
+        elsif action == :reinstate && !user.suspended?
+          log_texts << log_text[:user_is_not_suspended]
+        else
+          user.public_send("#{action}!")
```

Comment:
I'm always a little skeptical of `send` because it can make it hard to find usages of methods when searching source code. In this case, it's probably fine? The only alternatives I can think of:
- block form `proc { |user| user.reinstate! }
- a bunch more code duplication

Anyways something to consider if we ever add more of these or do more complicated interpolation

Reply:
Yeah, you are right it will be hard to read the code.